src/Backend.py

```diff
@@ -4,14 +4,14 @@
 
 import src.utilities as util
 import src.header as header
-from src.runcard_parsing import runcard_parsing
+import src.runmodes
 
 counter = None
 def init_counter(args):
     global counter
     counter = args
 
-class Backend(object):
+class Backend(src.runmodes.mode_selector[src.header.runmode]):
```

Comment:
I would do this dictionary selection outside of the class definition (even if just on top) because I think it is a bit cleaner. There are too many things in that single line.

Reply:
That makes sense. I'm not sure how dynamic inheritance works, and I was trying to get around ambiguities about orderings between when src.header.runmode is defined and when Backend.py is imported. Looking again, that might not be resolved by this anyway.

Long-term the best way could be to just move Backend.py to program_interface.py, and then wherever Backend instances are created, instead dynamically select a HEJ, NNLOJET instance instead, which inherit from Backend. For now I think a 'light touch method' is preferred though.